src/app/portlets/dot-edit-content/components/dot-edit-page-toolbar/dot-edit-page-toolbar.component.spec.ts

```diff
@@ -87,7 +130,95 @@ describe('DotEditPageToolbarComponent', () => {
+    });
+
+    it('should keep the locker true while moving through states', () => {
+        component.pageLocked = true;
+        fixture.detectChanges();
+
+        expect(component.stateSelected).toEqual('preview', 'preview state by default');
+
+        const stateSelectorButtons: DebugElement[] = de.queryAll(
+            By.css('.edit-page-toolbar__state-selector .ui-button')
+        );
+        const editButton = stateSelectorButtons[0].nativeElement;
+        const previewButton = stateSelectorButtons[1].nativeElement;
+        const liveButton = stateSelectorButtons[2].nativeElement;
+
+        let pageStateResult;
+        component.pageState.subscribe((res) => {
+            pageStateResult = res;
+        });
+
+        let lockPageResult;
+        component.lockPage.subscribe((res) => {
+            lockPageResult = res;
+        });
+
+        editButton.click();
+        expect(component.stateSelected).toEqual('edit', 'edit state selected');
+        expect(component.pageLocked).toBe(true, 'page locked after click in edit');
+        expect(pageStateResult).toEqual('edit', 'emit edit');
+        expect(lockPageResult).toBeUndefined('should not emit lock output');
+
+        previewButton.click();
+        expect(component.stateSelected).toEqual('preview', 'edit state selected');
+        expect(component.pageLocked).toBe(true, 'page locked after click in preview');
+        expect(pageStateResult).toEqual('preview', 'emit preview');
+        expect(lockPageResult).toBeUndefined('should not emit lock output');
+
+        liveButton.click();
+        expect(component.stateSelected).toEqual('live', 'live state selected');
+        expect(component.pageLocked).toBe(true, 'page locked after click in preview');
+        expect(pageStateResult).toEqual('live', 'emit live');
+        expect(lockPageResult).toBeUndefined('should not emit lock output');
+    });
+
+    it('should go to preview if user unlock the page while is in edit', () => {
+        fixture.detectChanges();
+
+        const lockerSwitch: DebugElement = de.query(By.css('.edit-page-toolbar__locker .ui-inputswitch'));
+
+        // Set the page locked and in edit mode
+        lockerSwitch.nativeElement.click();
+        expect(component.pageLocked).toBe(true, 'locked page');
+        expect(component.stateSelected).toEqual('edit', 'edit mode');
+
+        lockerSwitch.nativeElement.click();
+        expect(component.pageLocked).toBe(false, 'unlocked page');
+        expect(component.stateSelected).toEqual('preview', 'preview mode');
+    });
+
+    it('should go to edit if user lock the while is in preview', () => {
+        fixture.detectChanges();
+
+        const lockerSwitch: DebugElement = de.query(By.css('.edit-page-toolbar__locker .ui-inputswitch'));
+
+        lockerSwitch.nativeElement.click();
+        expect(component.pageLocked).toBe(true, 'locked page');
+        expect(component.stateSelected).toEqual('edit', 'edit mode');
```

Comment:
you should set the current state for 'preview' fisrt in this test

Reply:
Is preview by default at the component.